packages/react-components/react-progress/src/components/Progress/Progress.test.tsx

```diff
@@ -19,10 +19,26 @@ describe('Progress', () => {
     },
   });
 
-  // TODO add more tests here, and create visual regression tests in /apps/vr-tests
-
   it('renders a default state', () => {
     const result = render(<Progress>Default Progress</Progress>);
     expect(result.container).toMatchSnapshot();
   });
+  it('has role progressbar', () => {
+    const result = render(<Progress />);
+    expect(result.getByRole('progressbar')).toBeDefined();
+  });
+  it('doesnt add aria attributes for indeterminate', () => {
+    const result = render(<Progress />);
+    expect(result.getByRole('progressbar').getAttribute('aria-valuenow')).toBeFalsy();
+    expect(result.getByRole('progressbar').getAttribute('aria-valuemax')).toBeFalsy();
+    expect(result.getByRole('progressbar').getAttribute('aria-valuemin')).toBeFalsy();
+  });
+  it('adds aria attributes for determinate', () => {
+    const result = render(<Progress value={0.52} />);
+    expect(result.container.getElementsByClassName('fui-Progress__bar')[0].getAttribute('aria-valuenow')).toEqual(
```

Comment:
On the indeterminate test you're querying for the DOM element in a different way. I'd expect this test to use the same method as the other one. Is there a reason this is different?

Reply:
This is the correct way because the aria attributes are in the `bar` slot. I will update the `indeterminate` test to reflect this

Reply:
These aria attributes should be on the thing with `role="progressbar"`. So I think the implementation might need to be fixed to put them on the track, and this should be using `getByRole('progressbar')`.

Reply:
Generally it's a warning sign if you find yourself using something like `getElementsByClassName` in a test. See this document about queries in testing-library: https://testing-library.com/docs/queries/about/#priority

Reply:
Makes sense, I've made the changes!